src/mqt/bench/benchmark_generation.py

```diff
@@ -69,8 +69,8 @@ def generate_filename(
     benchmark_name: str,
     level: str,
     num_qubits: int | None,
-    gateset: Gateset | None = None,
-    device: Device | None = None,
+    gateset: Target | None = None,
+    device: Target | None = None,
```

Comment:
There is no need to distinguish between these anymore.
It is totally sufficient to have a single `target: Target | None = None` parameter.
The native gates level will just extract the gateset from the target.
The mapped level will use the full target.
The other levels do not use the information at all.

Reply:
Agreed.